lib/plenario_web/controllers/api/parse_column_params.ex

```diff
@@ -8,11 +8,30 @@ defmodule PlenarioWeb.Controllers.Api.CaptureArgs do
     |> Map.split(opts[:fields])
     |> elem(0)
     |> Stream.map(&split_value/1)
-    |> Enum.map(&to_condition_tuple/1)
+    |> Stream.map(&to_condition_tuple/1)
+    |> Enum.map(&cast/1)
     |> (&assign(conn, opts[:assign], &1)).()
   end
 
   defp split_value({col, val}), do: {col, String.split(val, ":", parts: 2)}
   defp to_condition_tuple({col, [val]}), do: {col, val}
   defp to_condition_tuple({col, [op, val]}), do: {col, {op, val}}
+
+  defp cast({col, {op, val}}) when is_binary(val) do
+    val = URI.decode(val)
+
+    val = case Poison.decode(val) do
+      {:error, {:invalid, _, _}} -> val
+      {:ok, map} -> 
+        try do
+          %{Geo.JSON.decode(map) | srid: 4326}
+        rescue
+          Geo.JSON.DecodeError -> map
+        end
+    end
+
+    {col, {op, val}}
+  end
+
+  defp cast({col, val}), do: {col, val}
```

Comment:
Jank bit of code. Binaries are automatically casted by ecto when building the queries later on. But it was extremely difficult to infer whether a binary represented a geojson map, or a time range map. In order to make make things easier for matching with `where_condition` (the function that generates the ecto query conditions), I do the casting here.